File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/model/erd/Tree.java
package org.javamaster.invocationlab.admin.model.erd;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * @author yudong
 * @date 2023/2/16
 */
@NoArgsConstructor
@Data
public class Tree {

    private String id;
    private String key;
    private String title;
    private String label;
    private String value;
    private Boolean isLeaf;
    private String parentId;
    private Object parentKey;
    private String sqlInfo;
    private String selectDB;
    private List<String> parentKeys;
    private List<Tree> children;

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/model/erd/Tree.kt
package org.javamaster.invocationlab.admin.model.erd

import org.javamaster.invocationlab.admin.annos.AllOpen

/**
 * @author yudong
 * @date 2023/2/16
 */


@AllOpen
class Tree {
    var id: String? = null
    var key: String? = null
    var title: String? = null
    var label: String? = null
    var value: String? = null
    var isLeaf: Boolean? = null
    var parentId: String? = null
    var parentKey: Any? = null
    var sqlInfo: String? = null
    var selectDB: String? = null
    var parentKeys: List<String>? = null
    var children: MutableList<Tree>? = null
}
